R/merge.R

```diff
@@ -81,9 +81,9 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       yy = y[missingyidx]
       othercolsx = setdiff(nm_x, by)
       if (length(othercolsx)) {
-        tmp = rep.int(NA_integer_, length(missingyidx))
-        # TO DO: use set() here instead..
-        yy = cbind(yy, x[tmp, othercolsx, with = FALSE])
+        nx = c(names(yy), paste0("V",seq_len(length(othercolsx))))
+        nx = make.unique(nx)
+        set(yy, NULL, tail(nx, -ncol(yy)), rep(list(NA), length(othercolsx)))
```

Comment:
does this need to be NA_integer_?

Reply:
Nope, `NA` logical works fine here. `rbindlist` coerces types so `logical` columns of `yy`will be coerced to the types of `dt`. The reason behind the previous `tmp` used `NA_integer_` is that it was used for indexing the data.table and therefore was `integer`.